src/plan_expand_hypertable.c

```diff
@@ -91,17 +91,13 @@ is_time_bucket_function(Expr *node)
 }
 
 static void
-ts_setup_append_rel_array(PlannerInfo *root)
+ts_add_append_rel_infos(PlannerInfo *root, List *appinfos)
 {
-	/* repalloc() does not work with NULL argument */
-	if (root->append_rel_array)
-		root->append_rel_array =
-			repalloc(root->append_rel_array, root->simple_rel_array_size * sizeof(AppendRelInfo *));
-	else
-		root->append_rel_array = palloc(root->simple_rel_array_size * sizeof(AppendRelInfo *));
-
 	ListCell *lc;
-	foreach (lc, root->append_rel_list)
+
+	root->append_rel_list = list_concat(root->append_rel_list, appinfos);
```

Comment:
Could you please add an assertion that `root->append_rel_list` is non-NULL and a comment that this is is set up in `expand_planner_arrays` so that it is clear that we expect that in the function. It should be properly set up in `expand_planner_arrays`, but if not, it is easier to understand what is the problem if it is NULL in this location if the precondition is clearly stated.

Reply:
root->append_rel_list is not required to be non-NULL
list_concat(NULL, list2) will return list_copy(list2)

Reply:
True, `list_concat` does not require it, but `expand_planner_array` always set it up, so it seems that if it is NULL here, something is not right.

Even with zero elements in the inheritance list inside `expand_inherited_rtentry`, it seems like it can call `expand_planner_arrays` with `add_size` being zero, which would mean that it is set up in that case anyway.

What am I missing?

Reply:
expand_planner_arrays does not touch root->append_rel_list, maybe you are confusing this with root->append_rel_array

Reply:
Ah, yes, I do! Sorry about that.